plugins/woocommerce/client/blocks/tests/e2e/tests/product-collection/product-collection-errors.block_theme.spec.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { Page } from '@playwright/test';
+import { test as base, expect, Editor, wpCLI } from '@woocommerce/e2e-utils';
+
+/**
+ * Internal dependencies
+ */
+
+class BlockUtils {
+	editor: Editor;
+	page: Page;
+
+	constructor( { editor, page }: { editor: Editor; page: Page } ) {
+		this.editor = editor;
+		this.page = page;
+	}
+
+	/**
+	 * Configures the Single Product Block in the editor.
+	 */
+	async configureSingleProductBlock( name?: string ) {
+		const singleProductBlock = await this.editor.getBlockByName(
+			'woocommerce/single-product'
+		);
+
+		if ( name ) {
+			await singleProductBlock
+				.locator( 'input[type="search"]' )
+				.fill( name );
+			await singleProductBlock.getByText( 'Search' ).click();
+			await singleProductBlock.getByText( name ).click();
+		} else {
+			await singleProductBlock
+				.locator( 'input[type="radio"]' )
+				.nth( 0 )
+				.click();
+		}
+
+		await singleProductBlock.getByText( 'Done' ).click();
+	}
+
+	async createManagedStockProduct() {
+		await wpCLI(
+			'wc product create --name="Managed Stock" --regular_price=10 --manage_stock=true --stock_quantity=1 --user=admin'
+		);
+	}
+}
+
+const test = base.extend< { blockUtils: BlockUtils } >( {
+	blockUtils: async ( { editor, page }, use ) => {
+		await use( new BlockUtils( { editor, page } ) );
+	},
+} );
+
+test.describe( 'Product Page: error notices when adding out-of-stock products', () => {
+	test( 'displays error notice when attempting to add product beyond stock limit', async ( {
+		admin,
+		editor,
+		blockUtils,
+		page,
+	} ) => {
+		await blockUtils.createManagedStockProduct();
+		await admin.createNewPost();
+		await editor.insertBlock( { name: 'woocommerce/single-product' } );
```

Comment:
We should be testing the _Product Collection_ block (`woocommerce/product-collection`) instead of the _Single Product_ block, here.